.code-samples.meilisearch.yaml

```diff
@@ -0,0 +1,494 @@
+settings_guide_searchable_1: |-
+  let searchable_attributes = &[
+    "uid",
+    "movie_id",
+    "title",
+    "description",
+    "poster",
+    "release_date",
+    "rank"
+  ];
+
+  let progress: Progress = movie_index.set_searchable_attributes(searchable_attributes).await.unwrap();
+settings_guide_displayed_1: |-
+  let displayed_attributes = &[
+    "title",
+    "description",
+    "poster",
+    "release_date",
+    "rank"
+  ];
+
+  let progress: Progress = movie_index.set_displayed_attributes(displayed_attributes).await.unwrap();
+settings_guide_accept_new_fields_1: |-
+  let progress: Progress = movie_index.set_accept_new_fields(false).await.unwrap();
+documents_guide_add_movie_1: |-
+  // Define the type of our documents
+  #[derive(Serialize, Deserialize, Debug)]
+  struct IncompleteMovie {
+    id: String,
+    title: String
+  }
+  impl Document for IncompleteMovie {
+    type UIDType = String;
+    fn get_uid(&self) -> &Self::UIDType { &self.id }
+  }
+
+  // Add a document to our index
+  let progress: Progress = movie_index.add_documents(&[
+    IncompleteMovie {
+      id: "123sq178".to_string(),
+      title: "Amelie Poulain".to_string(),
+    }
+  ], None).await.unwrap();
+search_guide_1: |-
+  // Build the query
+  let query: Query = Query::new("shifu")
+    .with_limit(5)
+    .with_offset(10);
+  
+  // Apply the query and get results
+  let results: SearchResults<Movie> = movie_index.search(&query).await.unwrap();
+search_guide_2: |-
+  // Build the query
+  let query: Query = Query::new("Avengers")
+    .with_filters("release_date > 795484800");
+  
+  // Apply the query and get results
+  let results: SearchResults<Movie> = movie_index.search(&query).await.unwrap();
+getting_started_create_index_md: |-
+  ```toml
+  [dependencies]
+  meilisearch-sdk = "0.2"
+  tokio = { version = "0.2", features = ["macros"] }  # required if you are not targeting wasm
+  serde = { version="1.0",   features = ["derive"] }  # required if you are going to use documents
+  serde_json = "1.0"                                  # required in some parts of this guide
+  ```
+
+  ```rust
+  use meilisearch_sdk::{indexes::*, document::*, client::*, search::*, progress::*, settings::*};
+  use serde::{Serialize, Deserialize};
+
+  #[tokio::main]
+  async fn main() {
+    let client = Client::new("http://localhost:7700", "masterKey");
+    // create an index if the index does not exist
+    let movie_index = client.get_or_create("movies").await.unwrap();  
+
+    // here is where you have to insert any other sample of this documentation
+  }
+  ```
+
+  Please read the second example below before any other example of this documentation.
+
+  [About this crate](https://github.com/meilisearch/meilisearch-rust)
+getting_started_add_documents_md: |-
+  The Rust library is quite different from the other libraries since documents are stronly typed.
+  You have to implement the `Document` trait on a struct to be able to use it with Meilisearch.
```

Comment:
No need to detail that from my POV, since other SDKs are also typed (go and .NET for example) 😉

Reply:
I like the second sentence but the first is not needed.